clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsGroupHeartbeatRequestManager.java

```diff
@@ -490,8 +492,12 @@ private NetworkClientDelegate.UnsentRequest makeHeartbeatRequestAndHandleRespons
     }
 
     private NetworkClientDelegate.UnsentRequest makeHeartbeatRequest(final long currentTimeMs) {
+        StreamsGroupHeartbeatRequestData data = this.heartbeatState.buildRequestData();
+        if (membershipManager.state() == MemberState.JOINING) {
+            logger.info("Joining group with request {}", data);
```

Comment:
No objections to this change, just unclear why we add it now?

Reply:
removed